torch/csrc/jit/api/object.h

```diff
@@ -102,6 +109,39 @@ struct TORCH_API Object {
     });
   }
 
+  // Used as a utility function to find a property
+  const std::vector<ObjectProperty> get_properties() const {
+    return c10::fmap(
+        type()->properties(), [&](const ClassType::Property& prop) {
+          c10::optional<Method> setter = c10::nullopt;
+          if (prop.setter) {
+            setter = Method(_ivalue(), prop.setter);
+          }
+          return ObjectProperty{
+              prop.name, Method(_ivalue(), prop.getter), setter};
+        });
+  }
+
+  bool has_property(const std::string& name) const {
+    auto props = get_properties();
+    for (auto prop : props) {
+      if (prop.name == name) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  const ObjectProperty get_property(const std::string& name) const {
+    auto props = get_properties();
```

Comment:
Rather than create `ObjectProperty` instances that you don't need, why not use `type->properties()` and only create the one `ObjectProperty` you actually need?

Alternatively, you can make all of the `ObjectProperty` instances once and then get the one corresponding to `name`.